### DISCUSSION TITLE
Signal to Noise ratio vs Elevation plot directly with Groops

### DISCUSSION THREAD

**Discussion by Bemtgenj** [groops-devs/groops#234] [Q&A]
Hello

First i like to wish everyone a happy new year.

I wanted to plot signal-to-noise ratio vs elevation directly from a GROOPS script and wanted to know if this is actually possible. I have done with Matlab but wanted to see if it is possible directly from GROOPS.
I have created the stn files per prn (as per printscreen). I think i did this correctly has i have one prn in the file (instead of multiple lines with different prn for same time stamp, -  the documentation states that {prn} is not necessary - so i am not 100% sure i used this input correctly).
 I did the same for the angle from the residual file but in the output there are multiple lines with the same time and the different prn. This then means it is not possible to merge the files for a given prn by taking columns S1C, S2L from one file and elevation from the other file. As the instrument files do not line up correctly (stn has one time, one prn, one S1C,one S2L per line and elevation has same time, multiple prn and then multiple azimuth / elevation lines)
Something like a "merge" column type of script would be needed with the condition on prn.

![image](https://github.com/groops-devs/groops/assets/100558965/e394755c-9374-498b-bbc6-3921d1ad07c3)

Thank you very much for the help

Cheers
Jean

**Comment by zhedumi** [comment]
Hi,

Yeah the input instrument files accomodate all PRNS. So if you want to showcase single PRNs I would just like you filter them for the specific PRNs within the observation instrumentfile and within the residual file get azimuth and elevation. 
After you have the PRN specific files I would reccomend to use [InstrumentSynchronize](https://groops-devs.github.io/groops/html/InstrumentSynchronize.html) incase of missing data points and set min gap = 0. 
After that you just need to use [InstrumentArcCalculate](https://groops-devs.github.io/groops/html/InstrumentArcCalculate.html) and just put in the inputfiles your two files. 
If the two files have the same data epochs than it will just merge them next to each other. Otherwise incase the sampling between observation and residual is different you might need to interpolate etc. but I presume that is not the case.

Cheers,
Patrick

**Reply by Bemtgenj** [reply]
Hi Patrick

Txs for the feedback - i must have made a mistake when i ran the azimuth / elevation instrument file in the first runs. It gave me some files with a specific prn (so correct output) and some with all the prns ("not correct output) which was then confusing me. I will try the other 2 steps mentioned above now. I might have to resample as i am not sure they all will be at the same data epochs

if i run into problems  i will let you know

Note: Maybe this is something that can be added to the documentation as is only states A1* and E1* can be used, but also files with specifying prns can be created

Cheers
Jean

**Comment by zhedumi** [comment]
Hi,

Yeah it might be a bit missleading to only use A1* and E1* since that just make the timeseries file for all PRN. I will ad a sentence to the documentation that will explicity say that if only a certain prn is to be used that E1*G10 for example should be used.

Cheers,

Patrick
